pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1531,6 +1531,72 @@ class Timestamp(_Timestamp):
             ) from err
         return _dt.ctime()
 
+    def date(self):
+        """
+        Return date object with same year, month and day.
+
+        Examples
+        --------
+        >>> ts = pd.Timestamp('2023-01-01 10:00:00.00')
+        >>> ts
+        Timestamp('2023-01-01 10:00:00')
+        >>> ts.date()
+        datetime.date(2023, 1, 1)
+        """
+        try:
+            _dt = datetime(self.year, self.month, self.day,
+                           self.hour, self.minute, self.second,
+                           self.microsecond, self.tzinfo, fold=self.fold)
+        except ValueError as err:
+            raise NotImplementedError(
+                "date not yet supported on Timestamps which "
+                "are outside the range of Python's standard library. "
+                "For now, please call the components you need (such as `.year` "
+                "and `.month`) and construct your string from there."
+            ) from err
+        return _dt.date()
+
+    def dst(self):
+        """
+        Return self.tzinfo.dst(self).
+
+        Examples
+        --------
+        >>> ts = pd.Timestamp('2000-06-01 00:00:00',
+        ...                   tz='Europe/Brussels').dst()
+        >>> ts
+        datetime.timedelta(seconds=3600)
+        """
+        _dt = datetime(self.year, self.month, self.day,
+                       self.hour, self.minute, self.second,
+                       self.microsecond, self.tzinfo, fold=self.fold)
+        return _dt.dst()
+
+    def isocalendar(self):
+        """
+        Return a named tuple containing ISO year, week number, and weekday.
+
+        Examples
+        --------
+        >>> ts = pd.Timestamp('2023-01-01 10:00:00')
+        >>> ts
+        Timestamp('2023-01-01 10:00:00')
+        >>> ts.isocalendar()
+        datetime.IsoCalendarDate(year=2022, week=52, weekday=7)
+        """
+        try:
+            _dt = datetime(self.year, self.month, self.day,
+                           self.hour, self.minute, self.second,
+                           self.microsecond, self.tzinfo, fold=self.fold)
+        except ValueError as err:
+            raise NotImplementedError(
+                "isocalendar not yet supported on Timestamps which "
+                "are outside the range of Python's standard library. "
+                "For now, please call the components you need (such as `.year` "
+                "and `.month`) and construct your string from there."
```

Comment:
I don't think this message is quite right, as users aren't constructing a string here

I think it's fine to leave out this part of the message here

Reply:
Done